### PR TITLE
feature(RR)

### PR DESCRIPTION
**Pull request by tejasai777**

### CODE REVIEW

File: TeamProject/TeamProject/main.cpp
Comment (alex-nieto) [inline TeamProject/TeamProject/main.cpp:1]: revert these changes

File: TeamProject/TeamProject/Simulation.cpp [hunk 1]
```diff
@@ -136,7 +135,7 @@ void Simulation :: runAlgorithm(Algorithm** algorithToRun, int arrSize){
     }
 }
 Process* Simulation :: copyProcessListToArr(){
-    Process newProcessArr[numOfProcesses];
+    Process* newProcessArr = new Process[numOfProcesses];
```
Comment (alex-nieto) [inline TeamProject/TeamProject/Simulation.cpp:138]: Why did you make this change?

File: TeamProject/TeamProject/Event.cpp [hunk 2]
```diff
@@ -7,69 +7,111 @@
...
+{
     return newState;
 }
-Process Event :: getProcess(){
+Process Event ::getProcess()
+{
     return process;
 }
-void Event :: printEventInfo(){
+void Event ::printEventInfo()
```
Comment (alex-nieto) [inline TeamProject/TeamProject/Event.cpp:87]: Again why did you do this? You should have saved the events in the events list so that way we print from the algo class.

File: TeamProject/TeamProject/main.cpp
Comment (tejasai777) [reply TeamProject/TeamProject/main.cpp:1]: @alex-nieto  will revert the changes just kept it for testing

File: TeamProject/TeamProject/Simulation.cpp (hunk 1 above)
Comment (tejasai777) [reply TeamProject/TeamProject/Simulation.cpp:138]: @alex-nieto  that's the one which throwing we an error while running thorugh i share you the ss right so i made those change. and it got resovled. it think it's creatign new processor insted of pass by refernce

File: TeamProject/TeamProject/Event.cpp (hunk 2 above)
Comment (tejasai777) [reply TeamProject/TeamProject/Event.cpp:87]: @alex-nieto  its just  a formatting thing

File: TeamProject/TeamProject/Simulation.cpp (hunk 1 above)
Comment (alex-nieto) [reply TeamProject/TeamProject/Simulation.cpp:138]: It needs to create a new one because if not all the alg will use same one and the updates from the others will still be there.

File: TeamProject/TeamProject/Event.cpp (hunk 2 above)
Comment (alex-nieto) [reply TeamProject/TeamProject/Event.cpp:87]: Oh nevermind. I am half asleep. I thought this was in the rr class

File: TeamProject/TeamProject/Algorithm.hpp [hunk 3]
```diff
@@ -39,7 +39,7 @@ class Algorithm{
     int calculateCPUutilization(); //calculates the cpu utilization for the algorithm
     //print methods
     void printAlgorithmInfo(); //outputs the algorithm info
-    void printProcessInfo(); //outputs the info for each process in the algorithm
+    virtual void printProcessInfo(); //outputs the info for each process in the algorithm
```
Comment (alex-nieto) [inline TeamProject/TeamProject/Algorithm.hpp:42]: This is what I was talking about why is needed

File: TeamProject/TeamProject/Simulation.cpp (hunk 1 above)
Comment (tejasai777) [reply TeamProject/TeamProject/Simulation.cpp:138]: ah you are right. okey in that case i need to update the process list at last so that its gonna print form the process list

File: TeamProject/TeamProject/Algorithm.hpp (hunk 3 above)
Comment (tejasai777) [reply TeamProject/TeamProject/Algorithm.hpp:42]: so when i update the process with finish time its not reflecting in the print statment

File: TeamProject/TeamProject/RR.cpp [hunk 4]
```diff
@@ -1,20 +1,110 @@
...
+        }
+    }
+}
+
+void RR::handleArrivalEvent(Event event)
+{
+    if (this->currentTime < event.getRemaningTime() < this->timeQuantam)
+    {
+        cout << "event arrived " << this->currentTime << " " << event.getProcess().getServiceTime() << "\n";
+        event.getProcess().setArrivalTime(this->currentTime);
```
Comment (tejasai777) [inline TeamProject/TeamProject/RR.cpp:78]: @alex-nieto  when i updated the set arrival time itsnot relfecting in the print statement
